File: leetcode/python/62_unique_paths.py
class Solution:

    def uniquePaths(self, cols: int, rows: int) -> int:
        curr = [1] + [0] * (cols-1)
        for _ in range(rows):
            for i in range(1, cols):
                curr[i] += curr[i-1]
        return curr[-1]

File: leetcode/java/62UniquePaths.java
class Solution {
  public long factorial(int number) {
    long result = 1;

    for (int factor = 2; factor <= number; factor++) {
      result *= factor;
    }

    return result;
  }

  public int uniquePaths(int m, int n) {
    return (int) (factorial(m + n - 2) / factorial(n - 1) / factorial(m - 1));
  }
}
